Adjust config path for FreeBSD compatibility

evitalis opened the pull request:
Tested on FreeBSD 11 without issue when making and installing from a package.
This should resolve #169

StephenBrown2 commented on contrib/holland-commvault/holland_commvault/commvault.py:
```diff
@@ -41,6 +41,13 @@ def main():
 
     argv = sys.argv[1:]
 
+    holland_conf = ''
```
Why not use the same logic as in the other files?
i.e. set holland_conf and then prepend /usr/local for freebsd

evitalis replied:
I didn't fix commvault on purpose as it is likely not used outside of specific environments. If @abg wants it fixed as well though I can.

StephenBrown2 commented on holland/core/cmdshell.py:
```diff
@@ -21,6 +21,11 @@
+holland_conf = '/etc/holland/holland.conf'
+currplatform = sys.platform
+if currplatform.startswith('freebsd'):
+    holland_conf = '/usr/local' + holland_conf
```
Should probably use [os.path.join()](https://docs.python.org/3/library/os.path.html#os.path.join) for this and the others.

evitalis replied:
os.path.join() cannot be used here as it will not join absolute paths. Since holland_conf is an absolute path this would fail unless it was adjusted that will then require and extra /  be added later.

abg replied:
I think using `sys.platform` directly is probably more idiomatic.  We're already using unixy paths, so maybe just use the simplest solution:

```
if sys.platform.startswith('freebsd'):
    holland_conf = '/usr/local/etc/holland/holland.conf'
else:
   holland_conf = '/etc/holland/holland.conf'
```

StephenBrown2 commented on holland/legacy/config.py:
```diff
@@ -25,8 +25,14 @@ def extract_params(section):
     global spec
+
+    holland_conf = '/etc/holland/holland.conf'
+    currplatform = sys.platform
```
Tiny nitpick, but maybe `curr_platform` or better, `current_platform` to match the other variables?

evitalis replied:
Typos on my part. I can fix it soon if needed, otherwise it does work as expected. I could even better yet just drop the variable and use sys.platform directly. Up to @abg and how he prefers the code be written I guess, at least as far as the PR is concerned.

abg replied:
This `holland/legacy/config.py` file is dead code and should be deleted.  Sloppy practices from a long time ago that never got cleaned up. :(

evitalis commented:
@abg and @StephenBrown2 I went ahead and fixed the checks and updated the code to be more simple. I left the legacy code alone since the commit was squashed, and it was stated that directory may be deleted.

Barring any other issues I think this should be good to go. Feel free to let me know or just fix anything I may have missed.

os.path.join() was still not feasible due to the second path being absolute. I also defined the default rather than rewrite the common part of the path in multiple spots.

evitalis commented:
@abg Can the version number be bumped when this is merged please? Will make it easier to finish my FreeBSD packages as well.

abg commented:
@evitalis Yeah, will do.  This looks good. Merging.   I'll bump the version and try to cut a new release soon.

StephenBrown2 commented:
@evitalis Yeah, I see the issue, but it could be remedied by stripping the slashes like so:
```
>>> os.path.join('/usr/local', holland_conf.strip('/'))
'/usr/local/etc/holland/holland.conf'
```
String concatenation works just as well.
